### DISCUSSION TITLE
Interface to AudioBus App for iOS

### DISCUSSION THREAD

**Discussion by vfantacone** [jamulussoftware/jamulus#3005] [Ideas]
iPhone/iPad AudioBus connection wanted!!… 
This feature would allow to use Samples Libraries on iPhone/iPdad with a Midi Controller for iOS Devices.
Very usefull if you are abroad and need to rehearse with your band mates.

**Comment by ann0see** [comment]
Hi.

Thanks for your feature request. Note that unfortunately we don't have an iOS developer in the main team and the iOS app doesn't really receive many feature updates.

@emlynmac created a native Jamulus  
client (called Jam it up) but I'm not sure about how far he got.

Otherwise KOORD actively develops the iOS app. @danryu

**Reply by vfantacone** [reply]
Does KOORD use Jamulus?

**Reply by ann0see** [reply]
Yes, they do.

**Reply by danryu** [reply]
Hi @vfantacone, yes at Koord.live we develop the Koord app as a clone of Jamulus, with various modifications (integrated ASIO driver, video integration via Koord forked-Jitsi service, integration with Koord.live private session service, modified GUI, App Store deployments etc).
As it's intended primarily for private session use, we remove the public directory browser, but otherwise the functionality is identical and you can use the client with your own sessions, public sessions etc (just without the video bit!).
By all means take a look at our iOS app, which is in beta state and available at https://koord.live/downloads (or search on the App Store).
_Note: all critical audio and network/data-processing codepaths are unchanged from Jamulus, in the Koord client app. The only difference currently is in the Windows ASIO driver initialization (incorporating our built-in open-source driver KoordASIO), and some fixes in Android device initialization._

Feel free to create an issue on our [Github issues page](https://github.com/koord-live/koord-app/issues) too. The repo is in an early state but we especially welcome end-users' suggestions for feature improvements :)
Specifically related to your request, we also currently have limited engineering resources, but are looking at moving the soundbase code to the JUCE framework, which supports IAA and AUv3 rather than AudioBus (as far as I know). Would IAA/AUv3 support your use case?
As I say, any issues on our tracker are welcome!

**Reply by vfantacone** [reply]
Yes, compatibility with apps like AUM or AudioBus would be OK.
The idea is to exploit the sount library of other apps like GarageBand, SampleTank, ecc directly in the iOS Device and use e.g. an iRig Midi Controller for plaing on KOORD/Jamulus.
I think this feature would boost the use of your app on iOS Devices!

**Reply by danryu** [reply]
It's a great suggestion. We look forward to supporting VST/AU plugin deployment in the near future :)

**Reply by emlynmac** [reply]
@vfantacone @danryu 
I have re-written the protocol layer for jamulus in pure Swift for the purpose of removing the licensing issues around the jamulus app and the Apple App stores. The work-in-progress is here: https://github.com/emlynmac/JamulusProtocol

As @ann0see said, I'm making an app called Jam It Up, which will essentially be a pure native client for apple devices and accessible through the app store. I've not had as much time as I wanted to work on it; it's an early alpha right now. 

I also want to use the the native implementation to enable Audio Units plugin capabilities, so that a jamulus stream could be dropped into Logic (or any other DAW supporting).
https://developer.apple.com/documentation/audiotoolbox/audio_unit_v3_plug-ins

If there's anything I can help with, just let me know - I don't have much time but might be able to help out here and there.
